jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/support/opentsdb/OpenTSDBMessageFormatter.java

```diff
@@ -101,7 +104,13 @@ public OpenTSDBMessageFormatter(@Nonnull ImmutableList<String> typeNames,
 	 */
 	void addTags(StringBuilder resultString) {
 		if (hostnameTag != null) {
-			addTag(resultString, "host", hostnameTag);
+			if (server != null) {
+				addTag(resultString, "host",
+						firstNonNull(server.getAlias(), server.getHost()));
```

Comment:
`firstNonNull(server.getAlias(), server.getHost()))` should be moved to the `Server` class. I'm not sure of the naming, but probably something like `getName()` or `getLabel()`. This same code is probably in quite a few other places.